In [1]:
import numpy as np
import pandas as pd
from scipy.stats import norm

Uma máquina produz perfumes com média de 150 ml e desvio padrão de 2 ml. Quatro frascos foram recolhidos ao acaso da produção e o volume médio desses 4 frascos foi 153,8 ml. Essa média permite dizer que a máquina está produzindo dentro da especificação? Realize o teste com nível de confinça de 95%.

H0: μ = 150 ml \
Ha: μ ≠ 150ml


Alpha = 1 - 0,95 = 0,05  \
Z calc = (Média observada - média populacional) / (desvio padrão / raiz de n)

In [5]:
z_calc = (153.8 - 150) / (2 / np.sqrt(4))
z_calc

3.8000000000000114

norm.pdf(x, loc=0, scale=1) -- Probability density function \
norm.cdf(x, loc=0, scale=1) -- Cumulative distribution function (área à esquerda de x) \
norm.sf(x, loc=0, scale=1) -- Survival function (1 - cdf) \
norm.ppf(x, loc=0, scale=1) -- Percent point function (inversa de cdf - percentil) \
norm.isf(x, loc=0, scale=1) -- Inverse survival function (inversa de sf)

In [6]:
norm.ppf(0.025)

-1.9599639845400545

In [7]:
norm.isf(0.025)

1.9599639845400545

In [9]:
norm.cdf(-1.96)

0.024997895148220435

In [10]:
norm.sf(1.96)

0.024997895148220435

Conclusão: \
Tendo em vista que z_calc = 3.8, que é maior que z_crit = 1.96, rejeitamos a hipótese nula. Portanto, a máquina NÃO está operando corretamente.